# HG changeset patch
# User Benjamin Lindner <lindnerb@users.sourceforge.net>
# Date 1223630014 -7200
#      Fri Oct 10 11:13:34 2008 +0200
# Branch release-3-0-x
# Node ID 1ffc52012b4670c1d1c08356ce6f5b1ca7bdba0f
# Parent  db621daaafaf6b6a6f9c77c410d955c9c477a1ca
enclose building direcries in quotes in pkg.m

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2008-10-02  Benjamin Lindner <lindnerb@users.sourceforge.net>
+
+	* pkg.m (configure_make): Enclose building directory in quotes.
+
 2008-09-29  Ben Abbott <bpabbott@mac.com>
 
 	* polynomial/deconv.m: Fix row/col orientation & length of output.
 
 	* polynomial/conv.m: Correct row/col orientation of output.
 
 2008-09-27  Ivan Sutoris  <ivan.sutoris@gmail.com>
 
diff --git a/scripts/pkg/pkg.m b/scripts/pkg/pkg.m
--- a/scripts/pkg/pkg.m
+++ b/scripts/pkg/pkg.m
@@ -1223,31 +1223,31 @@ function configure_make (desc, packdir, 
         flags = strcat (flags, " CXX=\"", octave_config_info ("CXX"), "\"");
       endif
       if (isempty (getenv ("AR")))
         flags = strcat (flags, " AR=\"", octave_config_info ("AR"), "\"");
       endif
       if (isempty (getenv ("RANLIB")))
         flags = strcat (flags, " RANLIB=\"", octave_config_info ("RANLIB"), "\"");
       endif
-      [status, output] = shell (strcat ("cd ", src, "; ./configure --prefix=\"",
+      [status, output] = shell (strcat ("cd '", src, "'; ./configure --prefix=\"",
                                         desc.dir, "\"", flags));
       if (status != 0)
 	rm_rf (desc.dir);
 	error ("the configure script returned the following error: %s", output);
       elseif (verbose)
 	printf("%s", output);
       endif
 
     endif
 
     ## make
     if (exist (fullfile (src, "Makefile"), "file"))
       [status, output] = shell (strcat ("export INSTALLDIR=\"", desc.dir,
-					 "\"; make -C ", src));
+ 					 "\"; make -C '", src, "'"));
       if (status != 0)
 	rm_rf (desc.dir);
 	error ("'make' returned the following error: %s", output);
       elseif (verbose)
 	printf("%s", output);
       endif
     endif
 
